src/support/modify.c

```diff
@@ -470,8 +470,7 @@ __wt_modify_reconstruct_from_upd_list(
     __wt_update_vector_init(session, &modifies);
     /* Find a complete update. */
     for (upd = modify; upd != NULL; upd = upd->next) {
-        if (upd->txnid == WT_TXN_ABORTED)
-            continue;
+        WT_ASSERT(session, upd->type != WT_UPDATE_TOMBSTONE);
 
         if (WT_UPDATE_DATA_VALUE(upd))
```

Comment:
Because we cannot see tombstone, this check can be simplified to only check a full value.

Reply:
I remember that there is code that we reconstruct the modifies from a tombstone by setting the buffer to empty value. We probably can remove that code as well.

Reply:
Also moved this to WT-12008.